model/sharing/setup.go

```diff
@@ -119,21 +119,20 @@ func (s *Sharing) Setup(inst *instance.Instance, m *Member) {
 		inst.Logger().WithNamespace("sharing").
 			Warnf("Error on setup replicate trigger (%s): %s", s.SID, err)
 	}
+	if s.FirstFilesRule() != nil {
+		if err := s.AddUploadTrigger(inst); err != nil {
+			inst.Logger().WithNamespace("sharing").
+				Warnf("Error on setup upload trigger (%s): %s", s.SID, err)
+		}
+	}
 	if err := s.InitialReplication(inst, m); err != nil {
 		inst.Logger().WithNamespace("sharing").
 			Warnf("Error on initial replication (%s): %s", s.SID, err)
 		s.retryWorker(inst, "share-replicate", 0)
 		if s.FirstFilesRule() != nil {
 			s.retryWorker(inst, "share-upload", 1) // 1, so that it will start after share-replicate
 		}
-	} else {
-		if s.FirstFilesRule() == nil {
-			return
-		}
-		if err := s.AddUploadTrigger(inst); err != nil {
-			inst.Logger().WithNamespace("sharing").
-				Warnf("Error on setup upload trigger (%s): %s", s.SID, err)
-		}
+	} else if s.FirstFilesRule() != nil {
```

Comment:
Since we don't return anymore when `s.FirstFileRule() == nil` we'll call `s.NotifyRecipients(inst, m)` anyway.
Is that a problem?

Reply:
No, it's not a problem.